drivers/adc-dac/ad5592r/ad5593r.c

```diff
@@ -83,30 +84,33 @@ int32_t ad5593r_write_dac(struct ad5592r_dev *dev, uint8_t chan,
 }
 
 /**
- * Read ADC channel.
+ * Read ADC.
  *
  * @param dev - The device structure.
- * @param chan - The channel number.
- * @param value - ADC value
+ * @param chans - The ADC channels to be readback
+ * @param values - ADC value array
  * @return 0 in case of success, negative error code otherwise
  */
-int32_t ad5593r_read_adc(struct ad5592r_dev *dev, uint8_t chan,
-			 uint16_t *value)
+int32_t ad5593r_read_adc(struct ad5592r_dev *dev, uint16_t chans,
+			 uint16_t *values)
 {
+	uint8_t data[AD5593R_ADC_VALUES_BUFF_SIZE], i, num_chans = 0;
 	int32_t ret;
-	uint8_t data[3];
-	uint16_t temp;
 
 	if (!dev)
 		return FAILURE;
 
-	temp = BIT(chan);
+	for (i = 0; i <= dev->num_channels; i++) {
```

Comment:
we use a thing called `hweight8` for achieving this, you would simplify the code a bit by getting rid of the loop and the `AD5592R_REG_ADC_SEQ_INCL` define.

Reply:
Good idea, this will require including the util library, is that okay?

Reply:
I would say yes, the stuff we have in util is pretty minimal, it shouldn't frighten anybody from using it. I also did a recursive grep in drivers after `util.h` and got 29 results.